packages/cli/src/commands/contract/call.ts

```diff
@@ -57,17 +95,94 @@ export class CallContract extends Command {
         `Deployment with timestamp ${deploymentData?.timestamp} has no deployment address!`
       );
 
-    execSync(
-      `cargo contract call --contract ${deploymentData.address} --message ${flags.message} ${
-        flags.args ? "--args " + flags.args : ""
-      } --suri //Alice --gas ${flags.gas ?? "100000000000"} --url ${resolveNetworkUrl(
-        config,
-        flags.network ?? ""
-      )} ${flags.dry ? "--dry-run" : ""}`,
-      {
-        stdio: "inherit",
-        cwd: path.resolve("contracts", contractInfo?.name ?? ""),
+    const accountData = config.accounts.find(
+      (account: AccountData) => account.alias === flags.account
+    );
+    if (!accountData) {
+      this.error("Provided account alias not found in swanky.config.json");
+    }
+
+    const mnemonic = accountData.isDev
+      ? (accountData.mnemonic as string)
+      : decrypt(
+          accountData.mnemonic as Encrypted,
+          (
+            await inquirer.prompt([
+              {
+                type: "password",
+                message: `Enter password for ${chalk.yellowBright(accountData.alias)}: `,
+                name: "password",
+              },
+            ])
+          ).password
+        );
+
+    const networkUrl = resolveNetworkUrl(config, flags.network ?? "");
+    console.log("AAAA", flags.params);
+    const api = new ChainApi(networkUrl);
+    await api.start();
+
+    const account = (await spinner.runCommand(async () => {
+      await cryptoWaitReady();
+      return new ChainAccount(mnemonic);
+    }, "Initialising")) as ChainAccount;
+
+    const metadata = (await spinner.runCommand(async () => {
+      if (!contractInfo.build) {
+        this.error(`No build info found for contract named ${args.contractName}`);
       }
+      const metadata = await readJSON(
+        path.resolve(contractInfo.build.artifactsPath, `${args.contractName}.json`)
+      );
+      return metadata;
+    }, "Getting metadata")) as Abi;
+
+    const contract = new ContractPromise(api.apiInst, metadata, deploymentData.address);
+
+    const storageDepositLimit = null;
+
+    const queryResult = await contract.query[flags.message](
+      account.pair.address,
+      {
+        gasLimit: -1,
+        storageDepositLimit,
+      },
+      ...flags.params
     );
+
+    if (args.messageType === "query") {
+      console.log(`Query result:`);
+      console.log(queryResult.result.toHuman());
```

Comment:
For now, it is okay, but this needs to be more human understandable, extracting necessary information only for example.

Reply:
agreed, we can discuss how to parse it for the next version.